sdk/python/feast/diff/registry_diff.py

```diff
@@ -77,14 +80,16 @@ def to_string(self):
 
 def tag_objects_for_keep_delete_update_add(
     existing_objs: Iterable[FeastObject], desired_objs: Iterable[FeastObject]
-) -> Tuple[Set[FeastObject], Set[FeastObject], Set[FeastObject], Set[FeastObject]]:
+) -> Tuple[List[FeastObject], List[FeastObject], List[FeastObject], List[FeastObject]]:
     existing_obj_names = {e.name for e in existing_objs}
+    desired_objs = list(desired_objs)
+    existing_objs = list(existing_objs)
     desired_obj_names = {e.name for e in desired_objs}
 
-    objs_to_add = {e for e in desired_objs if e.name not in existing_obj_names}
-    objs_to_update = {e for e in desired_objs if e.name in existing_obj_names}
-    objs_to_keep = {e for e in existing_objs if e.name in desired_obj_names}
-    objs_to_delete = {e for e in existing_objs if e.name not in desired_obj_names}
+    objs_to_add = [e for e in desired_objs if e.name not in existing_obj_names]
+    objs_to_update = [e for e in desired_objs if e.name in existing_obj_names]
+    objs_to_keep = [e for e in existing_objs if e.name in desired_obj_names]
+    objs_to_delete = [e for e in existing_objs if e.name not in desired_obj_names]
```

Comment:
Any reason for this? I think it may cause duplicates to be present in these lists.